app/services/workflow_executions/preparation_service.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'tempfile'
+
+module WorkflowExecutions
+  # Service used to Prepare a WorkflowExecution
+  class PreparationService < BaseService
+    def initialize(workflow_execution, user = nil, params = {})
+      super(user, params)
+      @workflow_execution = workflow_execution
+      @samplesheet_headers = %w[sample fastq_1 fastq_2]
+      @samplesheet_rows = []
+    end
+
+    def execute # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      # confirm params/permissions
+      # build workflow execution run directory
+      generate_run_dir
+
+      # does it make sense to rename the SamplesWorkflowExecution model to something better
```

Comment:
I think this depends on how we want to think of the SamplesWorkflowExecutions. Right now it's just a table join with extra samplesheet params and the name works fine for that, if we want it to handle more data or extra logic than a rename would make sense.

Reply:
I removed this comment for now, I think a rename will likely come into play when we decide how to store results for a Sample as this model reference might be the correct one.